db/dcrpg/internal/addrstmts.go

```diff
@@ -41,32 +41,43 @@ const (
 
 	SelectAddressAllByAddress = `SELECT * FROM addresses WHERE address=$1 order by block_time desc;`
 	SelectAddressRecvCount    = `SELECT COUNT(*) FROM addresses WHERE address=$1;`
-	SelectAddressesAllTxn     = `SELECT tx_hash, block_time as tx_time, ftxd.block_height as height 
-		from addresses left join transactions as ftxd on funding_tx_row_id=ftxd.id 
-		where address = $1 order by tx_time desc;`
-
-	SelectAddressesTxnByFundingTx = `SELECT tx_vin_vout_index, tx_hash, tx_vin_vout_index, 
-		block_height FROM addresses LEFT JOIN 
-		transactions on transactions.tx_hash=tx_hash and is_funding = FALSE WHERE 
-		address = $1 and tx_hash=$2;`
-
-	SelectAddressUnspentCountAndValue = `SELECT COUNT(*), SUM(value) FROM addresses 
+	SelectAddressesAllTxn     = `WITH these as (SELECT addresses.tx_hash, ftxd.time as tx_time, ftxd.block_height as height
+		from addresses left join transactions as ftxd on addresses.tx_hash=ftxd.tx_hash
```

Comment:
On the transactions table a `tx_hash` is only unique within a given block. This means we may have similar tx_hashes in multiple blocks. Therefore doing a left join on the `tx_hash` field might not be as effective as needed.
The only way I think you can fetch the information in the query above effectively is as below:
- Do a left join with vins table id & vouts table id  since they exist in both addresses as `tx_vin_vout_row_id` column and transactions table as `vin_db_ids` or `vout_db_ids` columns.
- Do a data migration to add `block_height` to  the addresses table.

Reply:
Actually doing a data migration to add the `block_height` back might help a lot with the left joins in the long run.

Reply:
`block_time` field already exists in the addresses table.

Reply:
If there is a realistic possibility of collisions for TxIDs then I will need to rebuild this since the strategy used is to get all the TxIDs ordered by block/time and then fill in the details for only the transactions that will be displayed.  The premise is that txids are unique.   

I just searched the db and we don't have any duplicate txids for is_valid=true transactions.  We do have 10 duplicates if we include is_valid=false transactions.

I'm not thrilled with this query for performance reasons, but given our desire to get testing underway with insight I think getting it merged in and learning what we don't know might be best.  I'm just pushing an update to only pull is_valid transactions.